openwisp_controller/config/models.py

```diff
@@ -198,6 +180,49 @@ def _auto_create_cert_extra(self, cert):
         return cert
 
 
+class Template(ShareableOrgMixin, AbstractTemplate):
+    """
+    openwisp-controller Template model
+    """
+    tags = TaggableManager(through='config.TaggedTemplate', blank=True,
+                           help_text=_('A comma-separated list of template tags, may be used '
+                                       'to ease auto configuration with specific settings (eg: '
+                                       '4G, mesh, WDS, VPN, ecc.)'))
+    vpn = models.ForeignKey('config.Vpn',
+                            verbose_name=_('VPN'),
+                            blank=True,
+                            null=True,
+                            on_delete=models.CASCADE)
+
+    # Define django_x509 concret model which will be
+    # use at the abstract model.
+    vpn_model = Vpn
+    ca_model = Ca
+    cert_model = Cert
+
+    class Meta(AbstractTemplate.Meta):
+        abstract = False
+        unique_together = (('organization', 'name'), )
+
+    def clean(self):
+        self._validate_org_relation('vpn')
+        if self.sharing == 'import':
+            data = self._get_remote_template_data()
+            import_org = Organization(**data['organization'])
```

Comment:
I don't understand, why are you creating an organization??

Reply:
@nemesisdesign this work was from another PR. it was needed here. it is not part of the logic to create external templates on the library. that is why it is executed for templates whose sharing setting is `import`. templates with `sharing settings = import` can't exist on the library. only public and shared_secret templates can. this was the import logic which admins will use on their different `openwisp_controller` instance to import the templates remotely.
I am creating the organization with which the template was imported with . that is if it doesn't already exist on that `openwisp_controller` instance.

Reply:
I still don't understand, I think we need to discuss this